### PR TITLE
Auto-update for packages related to 'healthcare-components-1.0.0-master-20201008-2'

### PR DESCRIPTION
**Pull request by MicrosoftHealthService**
Updates package 'Microsoft.Health.Abstractions' to version '1.0.0-master-20201008-2'
Updates package 'Microsoft.Health.Api' to version '1.0.0-master-20201008-2'
Updates package 'Microsoft.Health.Extensions.DependencyInjection' to version '1.0.0-master-20201008-2'
Updates package 'Microsoft.Health.Core' to version '1.0.0-master-20201008-2'
Updates package 'Microsoft.Health.SqlServer' to version '1.0.0-master-20201008-2'
Updates package 'Microsoft.Health.SqlServer.Api' to version '1.0.0-master-20201008-2'
Updates package 'Microsoft.Health.Extensions.BuildTimeCodeGenerator' to version '1.0.0-master-20201008-2'
Updates package 'Microsoft.Health.Test.Utilities' to version '1.0.0-master-20201008-2'
Updates package 'Microsoft.Health.Client' to version '1.0.0-master-20201008-2'

### CODE REVIEW

File: src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SearchParameterQueryGenerator.cs [hunk 1]
```diff
@@ -4,9 +4,9 @@
 // -------------------------------------------------------------------------------------------------
 
 using System;
-using System.Data.SqlClient;
 using System.Diagnostics;
 using System.Text.RegularExpressions;
+using Microsoft.Data.SqlClient;
```
Comment (rotodd) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SearchParameterQueryGenerator.cs:9]: Migrating to use Microsoft.Data.SqlClient (see PR [#104](https://github.com/microsoft/healthcare-shared-components/pull/104)).

File: src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs [hunk 2]
```diff
@@ -186,7 +186,7 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
                                                .AcceptVisitor(IncludeRewriter.Instance)
                                            ?? SqlRootExpression.WithDenormalizedExpressions();
 
-            using (SqlConnectionWrapper sqlConnectionWrapper = _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapper(true))
+            using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true))
```
Comment (rotodd) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs:189]: Changing calls to `SqlConnectionWrapperFactory` to be async (see PR [#102](https://github.com/microsoft/healthcare-shared-components/pull/102)).

File: src/Microsoft.Health.Fhir.SqlServer/Registration/FhirServerBuilderSqlServerRegistrationExtensions.cs [hunk 3]
```diff
@@ -22,12 +23,12 @@ namespace Microsoft.Extensions.DependencyInjection
...
     public static class FhirServerBuilderSqlServerRegistrationExtensions
     {
-        public static IFhirServerBuilder AddSqlServer(this IFhirServerBuilder fhirServerBuilder, Action<SqlServerDataStoreConfiguration> configureAction = null)
+        public static IFhirServerBuilder AddSqlServer(this IFhirServerBuilder fhirServerBuilder, IConfiguration configuration, Action<SqlServerDataStoreConfiguration> configureAction = null)
         {
             EnsureArg.IsNotNull(fhirServerBuilder, nameof(fhirServerBuilder));
             IServiceCollection services = fhirServerBuilder.Services;
 
-            services.AddSqlServerBase<SchemaVersion>(configureAction);
+            services.AddSqlServerBase<SchemaVersion>(configuration, configureAction);
```
Comment (rotodd) [inline src/Microsoft.Health.Fhir.SqlServer/Registration/FhirServerBuilderSqlServerRegistrationExtensions.cs:31]: Passing in config required for Managed Identity support (see PR [#102](https://github.com/microsoft/healthcare-shared-components/pull/102)).

File: src/Microsoft.Health.Fhir.SqlServer/Microsoft.Health.Fhir.SqlServer.csproj [hunk 4]
```diff
@@ -20,13 +20,13 @@
...
-    <PackageReference Include="Microsoft.Health.SqlServer.Api" Version="1.0.0-master-20200910-2" />
-    <PackageReference Include="Microsoft.Health.Extensions.BuildTimeCodeGenerator" Version="1.0.0-master-20200910-2" />
+    <PackageReference Include="Microsoft.Health.Abstractions" Version="1.0.0-master-20201008-2" />
+    <PackageReference Include="Microsoft.Health.SqlServer" Version="1.0.0-master-20201008-2" />
+    <PackageReference Include="Microsoft.Health.SqlServer.Api" Version="1.0.0-master-20201008-2" />
+    <PackageReference Include="Microsoft.Health.Extensions.BuildTimeCodeGenerator" Version="1.0.0-master-20201008-2" />
     <PackageReference Include="Microsoft.IO.RecyclableMemoryStream" Version="1.3.5" />
-    <PackageReference Include="Microsoft.SqlServer.SqlManagementObjects" Version="150.18208.0" />
     <PackageReference Include="Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions" Version="3.1.7" />
+    <PackageReference Include="Microsoft.SqlServer.SqlManagementObjects" Version="161.41011.9" />
```
Comment (rotodd) [inline src/Microsoft.Health.Fhir.SqlServer/Microsoft.Health.Fhir.SqlServer.csproj:29]: Upgrades `SqlManagementObjects` package to match version in PR [#104](https://github.com/microsoft/healthcare-shared-components/pull/104/).

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerFhirStorageTestsFixture.cs [hunk 5]
```diff
@@ -47,7 +49,7 @@ public SqlServerFhirStorageTestsFixture()
             var initialConnectionString = Environment.GetEnvironmentVariable("SqlServer:ConnectionString") ?? LocalConnectionString;
 
             _databaseName = $"FHIRINTEGRATIONTEST_{DateTimeOffset.UtcNow.ToUnixTimeSeconds()}_{BigInteger.Abs(new BigInteger(Guid.NewGuid().ToByteArray()))}";
-            string masterConnectionString = new SqlConnectionStringBuilder(initialConnectionString) { InitialCatalog = "master" }.ToString();
+            string masterConnectionString = new SqlConnectionStringBuilder(initialConnectionString) { InitialCatalog = MasterDatabaseName }.ToString();
```
Comment (rbans96) [inline test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerFhirStorageTestsFixture.cs:52]: It seems its not used anywhere

**Review by rbans96 — APPROVED** [review]

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerFhirStorageTestsFixture.cs (hunk 5 above)
Comment (rotodd) [reply test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerFhirStorageTestsFixture.cs]: Good catch, thanks! Removed in f50d036.

File: src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SearchParameterQueryGenerator.cs (hunk 1 above)
Comment (johnstairs) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SearchParameterQueryGenerator.cs:9]: Thanks Robin!

**Review by johnstairs — APPROVED** [review]

**Review by v-shaaal — APPROVED** [review]

**Pull request merged**